docs/deployment/cli.md

```diff
@@ -23,664 +23,254 @@ specific language governing permissions and limitations
-that will trigger a savepoint, and after the successful completion of that savepoint, they will finish by calling their
-`cancel()` method. If the `-d` flag is specified, then a `MAX_WATERMARK` will be emitted before the last checkpoint
-barrier. This will result all registered event-time timers to fire, thus flushing out any state that is waiting for
-a specific watermark, e.g. windows. The job will keep running until all sources properly shut down. This allows the
- job to finish processing all in-flight data.
-
-
-#### Cancel with a savepoint (deprecated)
+The path to the savepoint can be used later on to [restart the Flink job](#starting-a-job-from-a-savepoint).
 
-You can atomically trigger a savepoint and cancel a job.
+#### Disposing a Savepoint
 
+The `savepoint` action can be also used to remove savepoints. `--dispose` with the corresponding 
+savepoint path needs to be added:
 {% highlight bash %}
-./bin/flink cancel -s [savepointDirectory] <jobID>
+./bin/flink savepoint --dispose /tmp/flink-savepoints/savepoint-cca7bc-bb1e257f0dab $JOB_ID
 {% endhighlight %}
+```
+Disposing savepoint '/tmp/flink-savepoints/savepoint-cca7bc-bb1e257f0dab'.
+Waiting for response...
+Savepoint '/tmp/flink-savepoints/savepoint-cca7bc-bb1e257f0dab' disposed.
+```
+Triggering the savepoint disposal through the `savepoint` action does not only remove the data from 
+the storage but makes Flink clean up the savepoint-related metadata as well.
 
-If no savepoint directory is configured, you need to configure a default savepoint directory for the Flink installation (see [Savepoints]({% link ops/state/savepoints.md %}#configuration)).
-
-The job will only be cancelled if the savepoint succeeds.
-
-<p style="border-radius: 5px; padding: 5px" class="bg-danger">
-    <b>Note</b>: Cancelling a job with savepoint is deprecated. Use "stop" instead.
-</p>
-
-#### Restore a savepoint
+### Cancelling a Job
 
+Cancelling a job can be achieved through the `cancel` action:
 {% highlight bash %}
-./bin/flink run -s <savepointPath> ...
+./bin/flink cancel $JOB_ID
 {% endhighlight %}
+```
+Cancelling job cca7bc1061d61cf15238e92312c2fc20.
+Cancelled job cca7bc1061d61cf15238e92312c2fc20.
+```
+The corresponding job's state will be transitioned from `Running` to `Cancelled`. Any computations 
+will be stopped.
 
-The run command has a savepoint flag to submit a job, which restores its state from a savepoint. The savepoint path is returned by the savepoint trigger command.
-
-By default, we try to match all savepoint state to the job being submitted. If you want to allow to skip savepoint state that cannot be restored with the new job you can set the `allowNonRestoredState` flag. You need to allow this if you removed an operator from your program that was part of the program when the savepoint was triggered and you still want to use the savepoint.
+#### Stop: Cancelling a Job but creating a final Savepoint
```

Comment:
Here there is also the DRAIN option where we also emit the `MAX_WATERMARK`, right?